packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -159,18 +163,17 @@ class CreateCommand extends CreateBase {
     final bool metadataExists = projectDir.absolute.childFile('.metadata').existsSync();
     if (argResults['template'] != null) {
       template = stringToProjectType(stringArg('template'));
-    } else {
-      // If the project directory exists and isn't empty, then try to determine the template
-      // type from the project directory.
-      if (projectDir.existsSync() && projectDir.listSync().isNotEmpty) {
-        detectedProjectType = determineTemplateType();
-        if (detectedProjectType == null && metadataExists) {
-          // We can only be definitive that this is the wrong type if the .metadata file
-          // exists and contains a type that we don't understand, or doesn't contain a type.
-          throwToolExit('Sorry, unable to detect the type of project to recreate. '
-              'Try creating a fresh project and migrating your existing code to '
-              'the new project manually.');
-        }
+    }
+    // If the project directory exists and isn't empty, then try to determine the template
+    // type from the project directory.
+    if (projectDir.existsSync() && projectDir.listSync().isNotEmpty) {
+      detectedProjectType = determineTemplateType();
+      if (detectedProjectType == null && metadataExists) {
+        // We can only be definitive that this is the wrong type if the .metadata file
+        // exists and contains a type that we don't understand, or doesn't contain a type.
+        throwToolExit('Sorry, unable to detect the type of project to recreate. '
```

Comment:
You mean when the meta data file exists, but it does not contain the project type?

What is the desired behavior?

* no meta data file: allow any template passed in as argument
* meta data file but no project type: allow any template passed in as argument
* meta data file with project type: only allow identical template passed in as argument

Reply:
> You mean when the meta data file exists, but it does not contain the project type?

Sorry, yes; I misread the logic. That makes this less severe, but still not something we should break.

> What is the desired behavior?

If we can't determine what it was created with, we should use the provided template, as before your change.

Reply:
> meta data file but no project type

This actually already throws an error in the meta data parser.

>> What is the desired behavior?
>
> If we can't determine what it was created with

I would deduce intent from the code that is there:

1. Fail if the template in meta data and the template argument don't match up.
2. Fail if the template in meta data is unknown.

And then succeed in all other cases.

That seems better to me than leaving it as it was before the change, where it would just take the template argument always and never error on the argument and meta data being inconsistent. This is the behavior in the tool now.

If you prefer reverting it as to before this change, that is fine by me as well. Let me know what you prefer.